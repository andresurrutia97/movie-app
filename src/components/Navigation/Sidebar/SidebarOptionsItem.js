import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const SidebarOptionsItem = ({ path, name }) => {
  return (
    <NavLink
      activeClassName="active"
      className="flex items-center"
      to={path}
      exact
    >
      <div className="flag" />
      <span className="ml-5">{name}</span>
    </NavLink>
  );
};

SidebarOptionsItem.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default SidebarOptionsItem;
