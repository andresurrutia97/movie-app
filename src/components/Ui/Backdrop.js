import React from 'react';
import PropTypes from 'prop-types';

const Backdrop = ({ show, closed }) => {
  const baseBackdropClass = 'fixed w-full h-full bg-black opacity-50 z-30';

  const backdropClass = show
    ? [baseBackdropClass, 'block']
    : [baseBackdropClass, 'hidden'];

  return (
    <div
      className={backdropClass.join(' ')}
      role="presentation"
      onClick={closed}
    />
  );
};

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  closed: PropTypes.func,
};

Backdrop.defaultProps = {
  closed: () => {},
};

export default Backdrop;
