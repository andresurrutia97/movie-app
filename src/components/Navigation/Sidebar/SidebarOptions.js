import React from 'react';
import PropTypes from 'prop-types';
import { useAuth0 } from '@auth0/auth0-react';

import SidebarOptionsItem from './SidebarOptionsItem';

const SidebarOptions = ({ routes, close }) => {
  const { logout } = useAuth0();
  return (
    <ul className="w-full h-full mt-10 flex flex-col justify-between text-gray-300">
      <div>
        {routes.map(({ path, name }) => {
          return (
            <li
              className="item-sidebar"
              key={path}
              onClick={close}
              role="presentation"
            >
              <SidebarOptionsItem path={path} name={name} />
            </li>
          );
        })}
      </div>
      <li
        onClick={() => logout()}
        role="presentation"
        className="flex justify-between ml-2 itms-center text-gray-400 py-2 px-5 cursor-pointer"
      >
        Logout
        <span className="material-icons">exit_to_app</span>
      </li>
    </ul>
  );
};

SidebarOptions.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
  close: PropTypes.func.isRequired,
};

export default SidebarOptions;
