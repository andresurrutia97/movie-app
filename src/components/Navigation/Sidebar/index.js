import React from 'react';
import PropTypes from 'prop-types';

import Routes from '../../../routes';
import Backdrop from '../../Ui/Backdrop';
import SidebarOptions from './SidebarOptions';
import logo from '../../../img/logoWeb.png';

const Sidebar = ({ open, openHandler }) => {
  const baseSidebarClass =
    'h-full sidebar-width bg-black fixed py-6 transition-transform duration-300 ease-in-out z-30 md:hidden';

  const sidebarClass = open
    ? [baseSidebarClass, 'sidebar-open']
    : [baseSidebarClass, 'sidebar-close'];

  return (
    <>
      <Backdrop show={open} closed={openHandler} />
      <div className={sidebarClass.join(' ')}>
        <div className="h-full flex flex-col items-center">
          <img className="w-48" src={logo} alt="logo" />
          <SidebarOptions close={openHandler} routes={Routes} />
        </div>
      </div>
    </>
  );
};

Sidebar.propTypes = {
  open: PropTypes.bool.isRequired,
  openHandler: PropTypes.func.isRequired,
};

export default Sidebar;
