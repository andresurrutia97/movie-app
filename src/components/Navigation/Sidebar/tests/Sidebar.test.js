import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import Sidebar from '../index';
import Backdrop from '../../../Ui/Backdrop';
import SidebarOptions from '../SidebarOptions';

describe('<Sidebar/>', () => {
  const openHandler = jest.fn();

  const component = (
    <BrowserRouter>
      <Sidebar open={false} openHandler={openHandler} />
    </BrowserRouter>
  );

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an instance of a <Backdrop/> and a <SidebarOptions/>', () => {
    const wrapper = mount(component);

    expect(wrapper.contains(Backdrop)).toBeTruthy();
    expect(wrapper.contains(SidebarOptions)).toBeTruthy();
  });
});
