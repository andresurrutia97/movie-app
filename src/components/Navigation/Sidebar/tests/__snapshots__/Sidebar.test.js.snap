// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> should render correctly 1`] = `
Array [
  <div
    className="fixed w-full h-full bg-black opacity-50 z-30 hidden"
    onClick={[MockFunction]}
    role="presentation"
  />,
  <div
    className="h-full sidebar-width bg-black fixed py-6 transition-transform duration-300 ease-in-out z-30 md:hidden sidebar-close"
  >
    <div
      className="h-full flex flex-col items-center"
    >
      <img
        alt="logo"
        className="w-48"
        src="logoWeb.png"
      />
      <ul
        className="w-full h-full mt-10 flex flex-col justify-between text-gray-300"
      >
        <div>
          <li
            className="item-sidebar"
            onClick={[MockFunction]}
            role="presentation"
          >
            <a
              aria-current="page"
              className="flex items-center active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                Home
              </span>
            </a>
          </li>
          <li
            className="item-sidebar"
            onClick={[MockFunction]}
            role="presentation"
          >
            <a
              aria-current={null}
              className="flex items-center"
              href="/user"
              onClick={[Function]}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                Profile
              </span>
            </a>
          </li>
          <li
            className="item-sidebar"
            onClick={[MockFunction]}
            role="presentation"
          >
            <a
              aria-current={null}
              className="flex items-center"
              href="/toc"
              onClick={[Function]}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                ToC
              </span>
            </a>
          </li>
        </div>
        <li
          className="flex justify-between ml-2 itms-center text-gray-400 py-2 px-5 cursor-pointer"
          onClick={[Function]}
          role="presentation"
        >
          Logout
          <span
            className="material-icons"
          >
            exit_to_app
          </span>
        </li>
      </ul>
    </div>
  </div>,
]
`;
